Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -334,6 +347,97 @@ private Branch(Context context) {
 	}
 
 	/**
+	 *
+	 * <p>Creates and initialises a new  singleton instance of object of type {@link Branch} if
+	 * not existing.Application developers should not call this method directly.Instead extend
+	 * your Application class with {@link BranchLinkedApp} or use {@link BranchLinkedApp} as the
+	 * Application class in your manifest.
+	 * You need to set Branch key in your manifest</p>
+	 *
+	 * @see          <a href="https://github.com/BranchMetrics/Branch-Android-SDK/blob/05e234855f983ae022633eb01989adb05775532e/README.md#add-your-app-key-to-your-project">
+	 *                 Adding your app key to your project</a>
+	 *
+	 * @param application  Your {@link Application} instance.
+	 *
+	 */
+
+	public static void createInstance(Application application) {
+		isBranchAppInstantiated = application instanceof  BranchLinkApp;
+      /* Create and initialise a new instance of Branch. */
+		Branch createdInstance = getBranchInstance(application,true);
+
+		try {
+         /*Set an observer for ativity life cycle events*/
+			createdInstance.setActivityLifeCycleObserver(application);
+
+		}catch (NoSuchMethodError Ex){
+			//LifeCycleEvents are  available only from API level 14.
+			Log.w(TAG,BranchException.BRANCH_API_LVL_ERR_MSG);
+		}
+	}
```

Comment:
Can we merge this createInstance method into the existing `getBranchInstance()` method? something like
```
private static Branch getBranchInstance(Application application, boolean isLive) {
		if (branchReferral_ == null) {
			branchReferral_ = Branch.initInstance(application);

			String branchKey = branchReferral_.prefHelper_.getBranchKey(isLive);
	        if (branchKey == null || branchKey.equalsIgnoreCase(PrefHelper.NO_STRING_VALUE)) {
	        	Log.i("BranchSDK", "Branch Warning: Please enter your branch_key in your project's res/values/strings.xml!");
	        }
		}
		branchReferral_.context_ = application;
		
		isBranchAppInstantiated = application instanceof BranchLinkApp;
		
		try {
			/*Set an observer for ativity life cycle events*/
			branchReferral_.setActivityLifeCycleObserver(application);
			
		}catch (NoSuchMethodError Ex){
			//LifeCycleEvents are  available only from API level 14.
			Log.w(TAG,BranchException.BRANCH_API_LVL_ERR_MSG);
		}
		
		return branchReferral_;
	}
```